maps-app-ios/UI/Map View/MapViewController+GPS/MapViewController+LocationDisplay.swift

```diff
@@ -24,7 +24,9 @@ extension MapViewController {
         
         mapView.locationDisplay.autoPanModeChangedHandler = { newAutoPanMode in
             print("New autoPanMode: \(newAutoPanMode)")
-            self.gpsButton.setImage(self.mapView.locationDisplay.getImage(), for: .normal)
+            DispatchQueue.main.async {
```

Comment:
I realize the map is always displayed (and therefore never goes away?), but for the sake of correctness, the `autoPanModeChangedHandler` should weakly capture `self`.